### ISSUE TITLE
Make 64bit-compatible for foobar2000 version 2.0

### ISSUE THREAD

**Issue by stonecrusher** [zao/foo_wave_seekbar#45]
The [fb2k changelog](https://www.foobar2000.org/changelog) says
`64-bit versions require updated versions of add-on components.`

I would love to use wave_seekbar in the future with latest foobar. Any plans to make the shift to 64-bit?

**Comment by zao** [comment]
This has been requested both on IRC and in the forum thread.

I have not yet investigated how much work this would require and given the somewhat broken state the component was in when I stopped working on it, it could result in quite the support load to touch it again.

A possible rebuild would probably drop things like XP support and deprecate some functionality I cannot support anymore like old settings file formats and legacy compression schemes, requiring a full rescan of tracks.

**Comment by stonecrusher** [comment]
Sorry, I don't know about Forum or IRC. It's just important that you know people can't use the plugin in future versions.

XP support was dropped since foobar 1.6 anyway.
I think it's a good chance to get rid of old compability crutches anyway.

Here is (auto-translated) what another addon author (fbuser) thinks about the change:

> They "only" have to be recompiled. But this requires particularly with older components a certain adjustment need due to the different handling of pointers and integer variables for 32bit and 64bit programs.
> 
> For this it is necessary at least for some components (all the same whether 32bit or 64bit) to adapt these to the new configuration handling, so that configuration changes are not lost as before, if the application is terminated unexpectedly, e.g. by a crash.

**Comment by zao** [comment]
No worries, it's perfectly fine to raise this issue here. My intent was to note that you're not alone in asking about this as there has been voices in several other places.

The primary problem with building this component is that it requires a rather peculiar environment to build to begin with, making it brittle to any changes in library and toolchain versions.
As such it is a bit more work to modernise beyond adapting to the new SDK functionality that might have appeared since I last worked on it.

**Comment by ProTofik** [comment]
Hi. Has there been any update on this? If not, are any other similar components out there that can be used as an alternative to this one?

**Comment by grimes2** [comment]
https://www.foobar2000.org/components/view/foo_wave_minibar_mod

**Comment by zao** [comment]
Case's mod of my other component is a great alternative if it satisfies your requirements.

There's some progress behind the curtains, ported this component to D3D11 and it seems to work from the basic testing I've put it through.
Download and try it from https://github.com/zao/foo_wave_seekbar/releases/tag/0.3.0-alpha-1

**Comment by Renegade-Master** [comment]
> Case's mod of my other component is a great alternative if it satisfies your requirements.
> 
> There's some progress behind the curtains, ported this component to D3D11 and it seems to work from the basic testing I've put it through. Download and try it from https://github.com/zao/foo_wave_seekbar/releases/tag/0.3.0-alpha-1

Working for me too. I had forked the repo with the intention of helping, but it seems you pretty much have it working already. Thanks for your efforts!

I couldn't get the new D3D11 mode working though, but here is the old mode in the new Component which would be fine for most people I'm sure.
![image](https://github.com/zao/foo_wave_seekbar/assets/24991145/9689af8b-d7e5-4815-bacc-a0a7a2c52873)

**Comment by scaramousche** [comment]
@zao thank you for the alpha release; both GDI and D3D11 are working for me. Been scrambling around to restore component functionality after moving to foobar v2. Not knocking Cage's version at all, it’s v cool and all, but efficiency be damned I need those stereo waveforms.

**Comment by mzso** [comment]
@scaramousche Does it look different from the d3d9 scrollbar?

**Comment by scaramousche** [comment]
@mszo there are minute, pixel-level differences between versions.

the old version I was running: foobar 1.6.14, waveform seekbar 0.2.45 
on that version, there's no difference at all when switching between GDI and d3d9

current version I'm on: foobar 2.25.1, waveform seekbar 0.3.0 alpha 1
when I'm switching between d3d11 and GDI in it, d3d11 looks a little "fatter" and with less pronounced transients that GDI.

When selecting GDI in both foobar versions and switching quickly between them (obviously, with the same track loaded in both), I can probably see slight pixel-level differences between the two, but it’s on a massively nitpicky level. At a glance, d3d9, GDI and d3d11 all look basically the same. I only notice differences if I look really closely and switch back and forth between them.

**Comment by mzso** [comment]
@scaramousche 
I guess d3d9 wasn't working for you on 1.6.14, because that was always subtly different from GDI. Kind of how you describe the difference between GDI and d3d11.

**Comment by mzso** [comment]
@scaramousche So I tried it. But I downgraded immediately. The waveform with 0.3 is way more magnified. Rather squashes detail. Especially that even with old version the it goes to max much of the time with nowadays trendy dynamic range compressed music.

**Comment by scaramousche** [comment]
@mszo Could be that I'm not giving it enough space along the vertical axis to see a meaningful difference, or that I was checking on a track with rather sufficient dynamic range, or a combination of the two. But I'm not using the image for any analytical purpose, I have Reaper if I need that. I just prefer stereo music to look like stereo music when I'm listening and give me some visual indication of fun things going on, so the new version works perfectly for my scenario.

**Comment by mzso** [comment]
So for clarity's sake here's how it looks with 0.3:
<img width="740" height="53" alt="Image" src="https://github.com/user-attachments/assets/d7836eb6-346a-470c-a581-3e81339ce804" />
And with old version:

<img width="739" height="53" alt="Image" src="https://github.com/user-attachments/assets/c87b8ab8-6c79-48c8-a162-f8af60038000" />

And far worse with a "loudness war" DRC mastered music:
New:

<img width="740" height="53" alt="Image" src="https://github.com/user-attachments/assets/e6dc6cd5-4dc6-4f03-ad95-f3d38138fe29" />

Old:

<img width="739" height="53" alt="Image" src="https://github.com/user-attachments/assets/e335f5b0-bbb9-4ede-81fe-9aee81e14e20" />

Too bad the scaling can't be adjusted.

**Comment by scaramousche** [comment]
Yeah this doesn't look like my experience at all. My images are virtually identical between versions. So I checked dxdiag and wouldn't you know, my computer actually runs DirectX 12. So it’s possible that I'm not utilizing d3d11 at all even when the option's selected. But my waveforms still look very much like your old ones — I checked with the hottest masters that I think I have.
Sorry for leading you astray with my original assessment, if I have done.

**Comment by mzso** [comment]
@scaramousche Hm. Not sure if DX11 can be missing on W10/W11. But DX9 definitely needed to be installed manually.
Maybe you got caught out by the GDI fallback. I got that some years ago, when d3d9 wasn't working for some reason.
